litmus-portal/cluster-agents/subscriber/pkg/k8s/operations.go

```diff
@@ -243,7 +243,14 @@ func applyRequest(requestType string, obj *unstructured.Unstructured) (*unstruct
 		logrus.Info("successfully updated for kind: ", response.GetKind(), ", resource name: ", response.GetName(), ", and namespace: ", response.GetNamespace())
 		return response, nil
 	} else if requestType == "delete" {
-		err := dr.Delete(ctx, obj.GetName(), metav1.DeleteOptions{})
+		var err error
+		if obj.GetName() != "" {
+			err = dr.Delete(ctx, obj.GetName(), metav1.DeleteOptions{})
+			logrus.Info("successfully deleted for kind: ", obj.GetKind(), ", resource name: ", obj.GetName(), ", and namespace: ", obj.GetNamespace())
+		} else if obj.GetLabels() != nil {
+			err = dr.DeleteCollection(ctx, metav1.DeleteOptions{}, metav1.ListOptions{LabelSelector: convertMapToString(obj.GetLabels())})
```

Comment:
We can use `labels.FormatLabels()` instead of convertMapToString it's provided by apimachinery pkg

Reply:
yes, we can use but we need to ignore labels with  `executed_by` key.

Reply:
Hmm we could delete it first and then use the helper `delete(labels, "executed_by")`, it's probably better to use the official utils as much as possible also the code will be much shorter.